manager/controllers/app/m4dapplication_controller.go

```diff
@@ -73,10 +73,9 @@ func (r *M4DApplicationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
 			return result, err
 		}
 		applicationContext.Status.ObservedGeneration = applicationContext.GetGeneration()
-	} else if generatedBlueprint.Status.Ready {
-		applicationContext.Status.Ready = true
+	} else {
+		checkBlueprintStatus(applicationContext, generatedBlueprint)
```

Comment:
I would suggest leaving the ready field in the blueprint so that it's easy to look at it and understand its overall state.